File: frontend/src/app/request/CardSettingsRequest.ts
export class CardSettingsRequest {
  active: boolean;
  roundingStep: CardRoundingStep;
  cashback1: number;
  cashback2: number;
  cashback3: number;
  cashback4: number;

  constructor(active: boolean, roundingStep: CardRoundingStep, cashback1: number, cashback2: number, cashback3: number, cashback4: number) {
    this.active = active;
    this.roundingStep = roundingStep;
    this.cashback1 = cashback1;
    this.cashback2 = cashback2;
    this.cashback3 = cashback3;
    this.cashback4 = cashback4;
  }
}

enum CardRoundingStep {
  STEP10, STEP50,STEP100, STEP1000
}

File: backend/src/main/java/com/ifmo/payandsave/request/card/CardSettingsRequest.java
package com.ifmo.payandsave.request.card;

import com.ifmo.payandsave.model.card.enums.CardRoundingStep;
import lombok.*;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class CardSettingsRequest {
    private boolean active;
    private CardRoundingStep roundingStep;
    private Long cashback1;
    private Long cashback2;
    private Long cashback3;
    private Long cashback4;
}